src/essence/Tools/Measure/MeasureTool.js

```diff
@@ -1,858 +1,897 @@
-                scaleLng = d3
-                    .scaleLinear()
-                    .domain([segmentLengths[j], segmentLengths[j + 1]])
-                    .range([pts[j][0].y, pts[j][1].y])
-                //newProfileData.push({"x": scaleLat(cur), "y": scaleLng(cur), "value": });
-            }
-        }
-    }
-}
-
-function totalDistToIndex(l) {
-    var totalDistance = 0
-    for (var i = 1; i < l; i++) {
-        //Sum up segement distance
-        if (F_.dam) {
-            totalDistance += F_.distanceFormula(
-                clickedLatLngs[i].y,
-                clickedLatLngs[i].x,
-                clickedLatLngs[i - 1].y,
-                clickedLatLngs[i - 1].x
-            )
-        } else {
-            totalDistance += F_.lngLatDistBetween(
-                clickedLatLngs[i].y,
-                clickedLatLngs[i].x,
-                clickedLatLngs[i - 1].y,
-                clickedLatLngs[i - 1].x
-            )
-        }
-        //using actual latlng as meters:
-        //totalDistance += Math.sqrt(Math.pow(clickedLatLngs[i].x - clickedLatLngs[i-1].x, 2) + Math.pow(clickedLatLngs[i].y - clickedLatLngs[i-1].y, 2));
-    }
-    return totalDistance
-}
-
-MeasureTool.init()
-
-export default MeasureTool
+import $ from 'jquery'
+import * as d3 from 'd3'
+import F_ from '../../Basics/Formulae_/Formulae_'
+import L_ from '../../Basics/Layers_/Layers_'
+import Viewer_ from '../../Basics/Viewer_/Viewer_'
+import Map_ from '../../Basics/Map_/Map_'
+import Globe_ from '../../Basics/Globe_/Globe_'
+import CursorInfo from '../../Ancillary/CursorInfo'
+import calls from '../../../pre/calls'
+
+import metricsGraphics from '../../../external/MetricsGraphics/metricsgraphics.min'
+
+import ReactDOM from 'react-dom'
+import React, { useState, useEffect, useRef } from 'react'
+
+import { Line } from 'react-chartjs-2'
+import * as zoom from 'chartjs-plugin-zoom'
+
+import './MeasureTool.css'
+
+// Hacky solution to exposing setState externally
+let updateProfileData = function () {}
+let toolLayer = []
+let measureToolLayer = null
+let clickedLatLngs = []
+let distLineToMouse = null
+let distMousePoint = null
+let distDisplayUnit = "meters"
+let mode = 'segment'
+let steps = 100
+let profileData = []
+let elevPoints = []
+let profileDivId = 'measureToolProfile'
+let rAm = 100 //roundAmount
+
+const Measure = () => {
+    const [profileData, setProfileData] = useState([])
+    const refLine = useRef(null)
+
+    useEffect(() => {
+        updateProfileData = setProfileData
+        // code to run on component mount
+        Map_.map
+            .on('click', MeasureTool.clickMap)
+            .on('mousemove', MeasureTool.moveMap)
+            .on('mouseout', MeasureTool.mouseOutMap)
+
+        const globeCont = Globe_.litho.getContainer()
+        globeCont.addEventListener('click', MeasureTool.clickGlobe, false)
+        globeCont.addEventListener('mousemove', MeasureTool.moveGlobe, false)
+        development
```

Comment:
This line shouldn't be in here.

It also looks like the last commit removed a bunch of white spaces.